Panic using incorrectly quoted "queries" field key

### Bug report

__System info:__

* Version 0.12.0, branch 0.12, commit 4f29ba4013f6f46efeaa948bb7e0fd561cac5e95
* Ubuntu 14.04.4 LTS

__Steps to reproduce:__

1. Using Telegraf MySQL input plugin
2. Using CLI
3. Attempt to display data based on the `queries` field key
3. Screw up quotes on the query: `SELECT 'queries', slow_queries FROM ...`

__Expected behavior:__

Error because I should have used `"queries"` rather than `'queries'` 

__Actual behavior:__

```
[query] 2016/04/06 17:30:56 SELECT 'queries', slow_queries FROM telegraf."default".mysql WHERE time > now() - 1m
panic: unreachable

goroutine 127621 [running]:
github.com/influxdata/influxdb/influxql.buildAuxIterators(0xc21c8af8e0, 0x2, 0x2, 0x7f2401d879e8, 0xc20d1cbc60, 0x0, 0x0, 0xc21d9d7ba0, 0x1, 0x1, ...)
        /tmp/influxdb/src/github.com/influxdata/influxdb/influxql/select.go:125 +0x92f
github.com/influxdata/influxdb/influxql.Select(0xc20d1c4a90, 0x7f2401d879e8, 0xc20d1cbc60, 0xc20edb9b40, 0x0, 0x0, 0x0, 0x0, 0x0)
        /tmp/influxdb/src/github.com/influxdata/influxdb/influxql/select.go:49 +0x508
github.com/influxdata/influxdb/cluster.(*QueryExecutor).executeSelectStatement(0xc208072120, 0xc20d1c4a90, 0x2710, 0x0, 0x4614, 0xc213b8c0c0, 0xc21d9d4480, 0x0, 0x0)
        /tmp/influxdb/src/github.com/influxdata/influxdb/cluster/query_executor.go:500 +0x9c5
github.com/influxdata/influxdb/cluster.(*QueryExecutor).executeQuery(0xc208072120, 0xc217d46240, 0xc21c8a049b, 0x8, 0x2710, 0xc21d9d4480, 0xc213b8c0c0)
        /tmp/influxdb/src/github.com/influxdata/influxdb/cluster/query_executor.go:146 +0x9c0
created by github.com/influxdata/influxdb/cluster.(*QueryExecutor).ExecuteQuery
        /tmp/influxdb/src/github.com/influxdata/influxdb/cluster/query_executor.go:80 +0x8c
```

__Additional info:__

Using unquoted `queries` properly returns an error:

```
select queries, slow_queries from mysql where time > now() - 1m
ERR: error parsing query: found QUERIES, expected identifier, string, number, bool at line 1, char 8
```

Using double quoted `"queries"` returns valid data.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,7 @@
 ### Bugfixes
 
 - [#6206](https://github.com/influxdata/influxdb/issues/6206): Handle nil values from the tsm1 cursor correctly.
+- [#6248](https://github.com/influxdata/influxdb/issues/6248): Panic using incorrectly quoted "queries" field key.
 
 ## v0.12.0 [2016-04-05]
 ### Release Notes

--- a/influxql/select.go
+++ b/influxql/select.go
@@ -110,20 +110,27 @@ func buildAuxIterators(fields Fields, ic IteratorCreator, opt IteratorOptions) (
 
 	// Generate iterators for each field.
 	itrs := make([]Iterator, len(fields))
-	for i, f := range fields {
-		expr := Reduce(f.Expr, nil)
-		switch expr := expr.(type) {
-		case *VarRef:
-			itrs[i] = aitr.Iterator(expr.Val)
-		case *BinaryExpr:
-			itr, err := buildExprIterator(expr, aitr, opt)
-			if err != nil {
-				return nil, fmt.Errorf("error constructing iterator for field '%s': %s", f.String(), err)
+	if err := func() error {
+		for i, f := range fields {
+			expr := Reduce(f.Expr, nil)
+			switch expr := expr.(type) {
+			case *VarRef:
+				itrs[i] = aitr.Iterator(expr.Val)
+			case *BinaryExpr:
+				itr, err := buildExprIterator(expr, aitr, opt)
+				if err != nil {
+					return fmt.Errorf("error constructing iterator for field '%s': %s", f.String(), err)
+				}
+				itrs[i] = itr
+			default:
+				return fmt.Errorf("invalid expression type: %T", expr)
 			}
-			itrs[i] = itr
-		default:
-			panic("unreachable")
 		}
+		return nil
+	}(); err != nil {
+		Iterators(Iterators(itrs).filterNonNil()).Close()
+		aitr.Close()
+		return nil, err
 	}
 
 	// Background the primary iterator since there is no reader for it.

--- a/influxql/select_test.go
+++ b/influxql/select_test.go
@@ -2021,6 +2021,10 @@ func TestSelect_InvalidQueries(t *testing.T) {
 			q:   `SELECT 'value' FROM cpu`,
 			err: `invalid expression type: *influxql.StringLiteral`,
 		},
+		{
+			q:   `SELECT 'value', value FROM cpu`,
+			err: `invalid expression type: *influxql.StringLiteral`,
+		},
 	}
 
 	for i, tt := range tests {

